tests/OSModulesTest.php

```diff
@@ -98,9 +98,14 @@ public function testOS($os, $variant, $modules): void
         $this->requireSnmpsim();  // require snmpsim for tests
         // stub out Eventlog::log and Fping->ping, we don't need to store them for these tests
         $this->stubClasses();
+        // Order discovery modules
+        $modules = array_fill_keys(array_intersect(array_keys($this->discoveryModules), array_keys($modules)), true);
```

Comment:
Did you try without these? they should not be needed as the discovering and polling actually orders modules correctly 100% of the time now.

Reply:
Will do. 
And I'll open a dedicated PR if the QOS module orderBy is indeed fixing the error in [fee2cf1](https://github.com/librenms/librenms/pull/16856/commits/fee2cf1b7f7b42854332d5f63e12fa6d022b2446).

Reply:
QOS fix is good. PR on the way (and I remove it from here)